### PR TITLE
Refactoring the drop_counters test to reuse existing methods

### PR DESCRIPTION
**Pull request by vaibhavhd**
### Description of PR

Summary: Refactoring the existing methods as part of drop-counters testcase to be reused in measuring drops in dualtor testcases.
Fixes # https://github.com/Azure/sonic-mgmt/issues/2817

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Approach
Move the useful methods already present as part of drop-counter test to helpers dir. This is done so that these methods can now be used by other tests, such as dualtor testcases which require drop measurements.

#### What is the motivation for this PR?
Support dualtor testing.

#### How did you verify/test it?
Tested with a sample test:

```Python
def try_neighbor_flush(duthost):

    cmd_list = ["intfstat -D", "sonic-clear counters"]
    duthost.shell_cmds(cmds=cmd_list)

    logger.info("Output of L2 before: {}".format(get_pkt_drops(duthost, GET_L2_COUNTERS, None)))
    time.sleep(1)
    neigh_flush = duthost.shell("ip -4 neigh del 192.168.0.2 dev Vlan1000", module_ignore_errors=True)['stdout']
    logger.info("Output of action: {}".format(neigh_flush))
    time.sleep(5)
    logger.info("Output of L2 after: {}".format(get_pkt_drops(duthost, GET_L2_COUNTERS, None)))


def test_t1_to_server_active_tor_neigh_flush(duthosts,send_t1_to_server_after_action):
    active_host_before = duthosts[1]
    logger.info("Active ToR before {}".format(active_host_before.hostname))

    # perform bgp shutdown on active TOR
    send_t1_to_server_after_action(active_host_before,
                                action = lambda: try_neighbor_flush(active_host_before))
    # TODO add more verification steps here
#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation 
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

### CODE REVIEW

File: tests/common/helpers/drop_counters/drop_counters.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+GET_L3_COUNTERS = "intfstat -j "
+ACL_COUNTERS_UPDATE_INTERVAL = 10
+LOG_EXPECT_ACL_RULE_CREATE_RE = ".*Successfully created ACL rule.*"
+LOG_EXPECT_ACL_RULE_REMOVE_RE = ".*Successfully deleted ACL rule.*"
+LOG_EXPECT_PORT_ADMIN_DOWN_RE = ".*Configure {} admin status to down.*"
+LOG_EXPECT_PORT_ADMIN_UP_RE = ".*Port {} oper state set from down to up.*"
+RX_DRP = "RX_DRP"
+RX_ERR = "RX_ERR"
+
+PKT_NUMBER = 1000
```
Comment (daall) [inline tests/common/helpers/drop_counters/drop_counters.py:22]: I don't know how relevant this constant is to the dualtor testing - maybe it should stay in the `tests/drop_counters` folder? Should the below methods be updated to accept the number of expected drops as an argument?

**Review by neethajohn — COMMENTED** [review]
Please fix the LGTM as well

File: tests/common/helpers/drop_counters/drop_counters.py (hunk 1 above)
Comment (vaibhavhd) [reply tests/common/helpers/drop_counters/drop_counters.py]: Agree. Methods in this script are updated to expect the packets count as an argument.
Updated the `test_drop_counters` also accordingly.

File: tests/common/helpers/drop_counters/drop_counters.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        except ValueError as err:
+            logger.warning("Unable to verify L2 drops on iface {}\n{}".format(iface, err))
+            continue
+        if rx_drp_value >= PKT_NUMBER:
+            unexpected_drops[iface] = rx_drp_value
+    if unexpected_drops:
+        pytest.fail("L2 'RX_DRP' was incremented for the following interfaces:\n{}".format(unexpected_drops))
+
+
+def str_to_int(value):
```
[the remark this answers is not in the document]
Comment (vaibhavhd) [reply tests/common/helpers/drop_counters/drop_counters.py]: This was originally present in the test_drop_counter script. Deleted it as I could not find a usage.

**Comment by vaibhavhd** [comment]
> Please fix the LGTM as well

Fixed. Thanks!

**Review by neethajohn — APPROVED** [review]

**Pull request merged**